src/ashley/machina_extensions/forum_search/forms.py

```diff
@@ -13,6 +17,23 @@ class SearchForm(MachinaSearchForm):
     even with an empty search in the main field.
     """
 
+    def __init__(self, *args, **kwargs):
+        # load current lti_context to filter forum search
+        lti_contexts = kwargs.pop("lti_contexts", None)
+        super().__init__(*args, **kwargs)
+        user = kwargs.pop("user", None)
+        self.allowed_forums = PermissionHandler().get_readable_forums(
+            Forum.objects.filter(lti_contexts=lti_contexts), user
+        )
+        if self.allowed_forums:
+            self.fields["search_forums"].choices = [
+                (f.id, "{} {}".format("-" * f.margin_level, f.name))
+                for f in self.allowed_forums
+            ]
+        elif self.fields.get("search_forums", None) is not None:
+            # The user cannot view any single forum, the 'search_forums' field can be deleted
```

Comment:
I don't understand why you delete this information. As long as the forum is related to the current context, the user should be allowed to filter on it?

Reply:
I kept the original code of the base class MachinaSearchForm but only added my filter.
It deletes the field to know in search method in which case it is
the original code of the base class is this one 
`def __init__(self, *args, **kwargs):
        user = kwargs.pop('user', None)

        super().__init__(*args, **kwargs)

        # Update some fields
        self.fields['q'].label = _('Search for keywords')
        self.fields['q'].widget.attrs['placeholder'] = _('Keywords or phrase')
        self.fields['search_poster_name'].widget.attrs['placeholder'] = _('Poster name')

        self.allowed_forums = PermissionHandler().get_readable_forums(Forum.objects.all(), user)
        if self.allowed_forums:
            self.fields['search_forums'].choices = [
                (f.id, '{} {}'.format('-' * f.margin_level, f.name)) for f in self.allowed_forums
            ]
        else:
            # The user cannot view any single forum, the 'search_forums' field can be deleted
            del self.fields['search_forums']`

Reply:
In that case, thinking twice about it, the field is already deleted at this point as we call the init of MachinaSearchForm.

Reply:
In that case, thinking twice about it, the field is already deleted at this point as we call the init of MachinaSearchForm.